### PR TITLE
Fix: Optimized ItemParser

### PR DESCRIPTION
**Pull request by angelotadeucci**
It only takes 30 seconds now! Yoo-hoo!

### CODE REVIEW

File: GameDataParser/Parsers/ItemParser.cs [hunk 1]
```diff
@@ -16,8 +16,56 @@ public class ItemParser : Exporter<List<ItemMetadata>>
...
+                        if (!individualBoxItem.Attributes["locale"].Value.Equals("NA"))
+                        {
+                            continue;
+                        }
+                    }
+
+                    long box = long.Parse(individualBoxItem.Attributes["individualDropBoxID"].Value);
+                    int id = int.Parse(individualBoxItem.Attributes["item"].Value);
+
+                    if (individualBoxItem.Attributes["minCount"].Value.Contains("."))
```
Comment (AlanMorel) [inline GameDataParser/Parsers/ItemParser.cs:44]: this can be moved above the box and id parsing

File: GameDataParser/Parsers/ItemParser.cs [hunk 2]
```diff
@@ -16,8 +16,56 @@ public class ItemParser : Exporter<List<ItemMetadata>>
...
+                    int maxAmount = int.Parse(individualBoxItem.Attributes["maxCount"].Value);
+                    int dropGroup = int.Parse(individualBoxItem.Attributes["dropGroup"].Value);
+                    int smartDropRate = string.IsNullOrEmpty(individualBoxItem.Attributes["smartDropRate"]?.Value) ? 0 : int.Parse(individualBoxItem.Attributes["smartDropRate"].Value);
+                    int contentRarity = string.IsNullOrEmpty(individualBoxItem.Attributes["PackageUIShowGrade"]?.Value) ? 0 : int.Parse(individualBoxItem.Attributes["PackageUIShowGrade"].Value);
+                    int enchant = string.IsNullOrEmpty(individualBoxItem.Attributes["enchantLevel"]?.Value) ? 0 : int.Parse(individualBoxItem.Attributes["enchantLevel"].Value);
+                    int id2 = string.IsNullOrEmpty(individualBoxItem.Attributes["item2"]?.Value) ? 0 : int.Parse(individualBoxItem.Attributes["item2"].Value);
+
+                    if (itemDrops.ContainsKey(box))
+                    {
+                        itemDrops[box].Add(new ItemContent(id, minAmount, maxAmount, dropGroup, smartDropRate, contentRarity, enchant, id2));
```
Comment (AlanMorel) [inline GameDataParser/Parsers/ItemParser.cs:58]: Create the instance of ItemContent before the if statement, then use it here and on line 62

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Review by Sparkymod — APPROVED** [review]
amazing

File: GameDataParser/Parsers/ItemParser.cs [hunk 3]
```diff
@@ -16,8 +16,57 @@ public class ItemParser : Exporter<List<ItemMetadata>>
...
+                            continue;
+                        }
+                    }
+
+                    if (individualBoxItem.Attributes["minCount"].Value.Contains("."))
+                    {
+                        continue;
+                    }
+
+                    long box = long.Parse(individualBoxItem.Attributes["individualDropBoxID"].Value);
```
Comment (rhydev) [inline GameDataParser/Parsers/ItemParser.cs:46]: I don't think we need to parse to long here, can just keep as string

**Review by rhydev — CHANGES_REQUESTED** [review]

**Review by rhydev — APPROVED** [review]

**Pull request merged**